src/dosext/dpmi/dpmi.c

```diff
@@ -397,6 +397,8 @@ int dpmi_is_valid_range(dosaddr_t addr, int len)
     return 0;
   if (blk->base + blk->size < addr + len)
     return 0;
+  if (is_hardware_ram(addr, len))
```

Comment:
dpmi_is_valid_range() is used mostly in dump
routines to show ops and stack.
It would be strange to print ops and stack
from hw ram.
You can add another func perhaps.

Reply:
ok, will do that in another PR. For now I've just applied the fix and optimized it.